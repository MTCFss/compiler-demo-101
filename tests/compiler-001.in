if (1) {
  int lol, olo; // lol

  signed /*COMMET
       COMMENT2 */ int long long lol = 5 + 9 - 2 + 020 * - 0xF2;

  5;
  char e, e1 = ('a'), e2 = '\'', e3;

  unsigned float f = (5.2 * 6) + 9;

  char test = "lol"
              "olo";

  { 5 + 6; }

  lol = -6e5 + 6. + 6.3 - -6.3e10;
  int T[ 5 ];
  int r = T[5];
  // int r = 5 + T[5];
  +-+-+-+-+-+-+ +-+-+-+-+-+- -+-+-+-+ - 5;
  'Z' - 'a';
  cos ( 5 );
  log(4);
  log2((2 * 2 * 2));
  // ((((x)))) = ((((6 - ((((6))))))));
  x = ((((6 - ((((6))))))));
}
