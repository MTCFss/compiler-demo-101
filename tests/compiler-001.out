NUMBER<1> 
VALUE<1>     
TYPE<int>
ID<lol>


ID<olo>
TYPE_SPECIFIER<signed>

TYPE_SPECIFIER<long>
TYPE_SPECIFIER<long>
TYPE<int>
NUMBER<5>
NUMBER<9>
NUMBER<2>
NUMBER<16>
  NUMBER<242>
OP<*>:1:1:1:1
OP<+>:1:1:1:1
OP<->:1:1:1:1
		OP<+>:1:1:1:1	 

VALUE<3884>
ID<lol>
NUMBER<5>
VALUE<5>
TYPE<char>
ID<e>
CHAR<a>
VALUE<a>
ID<e1>
CHAR<\>
VALUE<\>
ID<e2>
ID<e3>
TYPE_SPECIFIER<unsigned>
TYPE<float>
FLOAT<5.2>
NUMBER<6>
OP<*>:1:1:1:1
NUMBER<9>
OP<+>:1:1:1:1
VALUE<40.2>
ID<f>
TYPE<char>
STRING<lol>
STRING<olo>
VALUE<lololo>
ID<test>
NUMBER<5>
NUMBER<6>
OP<+>:1:1:1:1
VALUE<11>
FLOAT<600000>
FLOAT<6>
FLOAT<6.3>
FLOAT<6.3e+10>
OP<->:1:1:1:1
OP<+>:1:1:1:1
OP<+>:1:1:1:1
VALUE<-6.30006e+10>
TYPE<int>
NUMBER<5>
ARRAY<T,5>
TYPE<int>
NUMBER<5>
ID<r>
NUMBER<5>
VALUE<-5>
CHAR<Z>
CHAR<a>
OP<->
VALUE<ù>
NUMBER<5>
FNCT<cos(5),0.283662>
VALUE<0.283662>
NUMBER<4>
FNCT<log(4),1.38629>
VALUE<1.38629>
NUMBER<2>
NUMBER<2>
NUMBER<2>
OP<*>:1:1:1:1
OP<*>:1:1:1:1
FNCT<log2(8),3>
VALUE<3>
NUMBER<6>
NUMBER<6>
OP<->:1:1:1:1
VALUE<0>


